### PR TITLE
Document all features on docs.rs

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

In particular we weren't generating documentation for flight-sql

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-flight/Cargo.toml [hunk 1]
```diff
@@ -40,6 +40,9 @@ prost-derive = { version = "0.11", default-features = false }
 tokio = { version = "1.0", default-features = false, features = ["macros", "rt", "rt-multi-thread"] }
 futures = { version = "0.3", default-features = false, features = ["alloc"]}
 
+[package.metadata.docs.rs]
```
Comment (alamb) [inline arrow-flight/Cargo.toml:43]: 👍

File: arrow-ord/Cargo.toml [hunk 2]
```diff
@@ -48,6 +48,9 @@ num = { version = "0.4", default-features = false, features = ["std"] }
 [dev-dependencies]
 rand = { version = "0.8", default-features = false, features = ["std", "std_rng"] }
 
+[package.metadata.docs.rs]
+features = ["dyn_cmp_dict"]
```
Comment (alamb) [inline arrow-ord/Cargo.toml:52]: this can't be `simd` because it requires nightly?

**Review by alamb — APPROVED** [review]
👍  Thank you @tustvold . I don't really know how to test this 

See also https://github.com/apache/arrow-rs/pull/3372

File: arrow-ord/Cargo.toml (hunk 2 above)
Comment (tustvold) [reply arrow-ord/Cargo.toml:52]: Tbh I'm not 100% sure, I'm basing this off the top-level crate and erring on the side of caution as I don't know how to test this 😅

**Pull request merged**